### PR TITLE
[WIP] Add metrics for scheduling

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/controllers/scheduling/metrics.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	}
+}
+
+func (m *scheduleMetrics) startSchedule(key string) {
+	if m == nil {
+		return
+	}
+
+	m.scheduleStartTimes[key] = m.clock.Now()
+	if _, exists := m.scheduleStartTimes[key]; !exists {
```
Comment (xuezhaojun) [inline pkg/controllers/scheduling/metrics.go:75]: It feels `m.scheduleStartTimes[key]` must exist after the previous line.

File: pkg/controllers/scheduling/metrics.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	BindDurationKey       = "bind_duration_seconds"
+)
+
+var (
+	schedulingDuration = k8smetrics.NewHistogramVec(&k8smetrics.HistogramOpts{
+		Subsystem:      SchedulingSubsystem,
+		Name:           SchedulingDurationKey,
+		StabilityLevel: k8smetrics.ALPHA,
+		Help:           "How long in seconds schedule a placement.",
+		Buckets:        k8smetrics.ExponentialBuckets(10e-9, 10, 10),
```
Comment (elgnay) [inline pkg/controllers/scheduling/metrics.go:23]: The exponential buckets look like
```
# TYPE scheduling_bind_duration_seconds histogram
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="1e-08"} 0
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="1e-07"} 0
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="1e-06"} 0
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="9.999999999999999e-06"} 1
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="9.999999999999999e-05"} 95
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="0.001"} 97
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="0.01"} 116
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="0.1"} 126
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="1"} 126
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="10"} 126
scheduling_bind_duration_seconds_bucket{name="placement_scheduling",le="+Inf"} 126
scheduling_bind_duration_seconds_sum{name="placement_scheduling"} 0.32492658600000013
scheduling_bind_duration_seconds_count{name="placement_scheduling"} 126
```
How about linear buckets?

**Comment by elgnay** [comment]
When the histogram data be reset? for example, if all placements are removed, will the data be updated or not?

File: pkg/controllers/scheduling/metrics.go (hunk 2 above)
Comment (qiujian16) [reply pkg/controllers/scheduling/metrics.go:23]: seems to me exponential buckets makes more sense? we care about the placement scheduled less then 1 seconds, and then 10 seconds

**Pull request closed without merging**